### ISSUE TITLE
Curb leaking sockets on Windows

### ISSUE THREAD

**Issue by vais** [TinderBox/dynamics_crm#22]
After about 50 requests to CRM, the Ruby process starts failing with "Too many open files" error. I have only experienced this on Windows, and suspect this is a Windows-only issue with Curb. Even though the `post` method in client.rb calls `close` on the `Curl::Easy` instance, it seems to have no effect, and looking at `netstat -a` output you can see connections building up with each call and never closed. I tried every possible way to get Curb to close its connections to no avail.

In the end I solved this problem for myself by rewriting the `post` method using net/http as follows:

```
    require 'net/http'
    def post(url, request)
      log_xml("REQUEST", request)

      uri = URI(url)
      ssl = {
        :use_ssl => (uri.scheme == 'https'),
        :verify_mode => OpenSSL::SSL::VERIFY_NONE
      }

      response = Net::HTTP.start(uri.host, uri.port, ssl) do |http|
        req = Net::HTTP::Post.new(uri)
        req['Connection'] = 'Keep-Alive'
        req['Content-type'] = 'application/soap+xml; charset=UTF-8'
        req.body = request
        http.request(req).body
      end

      log_xml("RESPONSE", response)
      response
    end
```

One might say I kicked Curb to the curb :)

Since this is the only place in this gem that actually uses Curb, it is possible to drop the dependency on Curb entirely. The only reservation I have is that net/http does not follow redirects automatically. I see that in the original `post` method you use the `follow_location` feature of Curb. My experience with CRM is entirely read-only, and there has not been any need for following redirects so far. So, I don't know if CRM ever responds with a redirect - do you know of some use cases where it does? If not, I'd be happy to submit a pull request to get rid of Curb dependency in this gem. It was such a PITA to get it to work on Windows in the first place, and now this thing. I would be happy to see Curb gone :)

**Comment by jheth** [comment]
Hi @vais, thanks for reporting this issue, I had not attempted to use this on Windows yet.  What were the other issues you ran into?  I believe there is an authentication workflow that needed to support redirects, I'll have to revisit when/why that was added. I like the idea of removing another dependency from gem.

**Comment by vais** [comment]
> What were the other issues you ran into?

Aside from this one, I have not run into any other issues so far (not counting the [initial installation difficulties](https://github.com/TinderBox/dynamics_crm/wiki/Windows-Installation-Gotchas)).

> I believe there is an authentication workflow that needed to support redirects, I'll have to revisit when/why that was added. I like the idea of removing another dependency from gem.

:+1:

**Comment by jheth** [comment]
This will be resolved by #54
